internal/importers/base/importer.go

```diff
@@ -251,6 +251,12 @@ func (i *Importer) ImportPackages(packages []ImportedPackage, defaultConstraintF
 			pc.Constraint = gps.Any()
 		}
 
+		// Skip constraints that are empty (no branch, version or source)
+		if gps.IsAny(pc.Constraint) && pc.Ident.Source == "" {
+			i.Logger.Printf("  Skipping import of %v because its constraint is empty.\n", pc.Ident)
+			continue
+		}
```

Comment:
We should still add an entry to the lock. Let's drop the continue in favor of wrapping the following lines that handle adding the constraint to the manifest instead:

https://github.com/golang/dep/blob/b005525be34491c2ffba98382a2e69c09efad405/internal/importers/base/importer.go#L254-L258